### ISSUE TITLE
await device.takeScreenshot to return the image

### ISSUE THREAD

**Issue by wilau2** [wix/Detox#1362]
`await device.takeScreenshot('MyScreenShot');`
could return the screenshot image. 
This would facilitate a lot visual regression testing 

```
const image = await device.takeScreenshot('MyScreenShot');
jestExpect(image).toMatchImageSnapshot();
```

```init.js

const detox = require('detox');
const config = require('../../package.json').detox;
const adapter = require('detox/runners/jest/adapter');
const jestExpect = require('expect');
const { toMatchImageSnapshot } = require('jest-image-snapshot');
jestExpect.extend({ toMatchImageSnapshot });

global.jestExpect = jestExpect;

jest.setTimeout(120000); // 2 minutes
jasmine.getEnv().addReporter(adapter);

beforeAll(async () => {
  await detox.init(config);
});

beforeEach(async () => {
  await adapter.beforeEach();
});

afterAll(async () => {
  await adapter.afterAll();
  await detox.cleanup();
});
```

**Comment by wilau2** [comment]
Workaround :
```
const fs = require('fs');
const util = require('util');
const readFile = util.promisify(fs.readFile);
const { exec } = require('child-process-async');

const takeScreenShot = async (storyname) => {
  await exec(`xcrun simctl io booted screenshot /tmp/${filename}.png`);
  const image = await readFile(`/tmp/${filename}.png`);
  return image;
};

describe('Visual regression', () => {
  it('should match screenshot', async () => {
    const image = await takeScreenShot(`test`);
    jestExpect(image).toMatchImageSnapshot();
  });
});
```

**Comment by LeoNatan** [comment]
This can potentially have memory issues. Loading a huge iPad Pro or iPhone XS Max screenshots in memory would be slow and memory intensive.

**Comment by wilau2** [comment]
Can we add disclaimers of usages in the documentation?
The syntax: 
``
const image = await device.takeScreenshot('MyScreenShot');
jestExpect(image).toMatchImageSnapshot();
```
Makes it super easy and clean for the developer. It can also benefit from the update snapshot jest command. 

The bad thing about my workaround is that I don't get the deviceId  
`xcrun simctl io booted screenshot /tmp/${filename}.png`

**Comment by LeoNatan** [comment]
You can find the current device id in Detox somewhere. Screenshot comparison is a low priority for us. I’d rather we conserve memory and CPU time.

**Comment by noomorph** [comment]
@LeoNatan, I think that we might facilitate getting **a resolved path to the screenshot**.
If Detox users then decide to load&compare huge images (aka burn some memory and CPU time), then it will be entirely their responsibility.

**Comment by LeoNatan** [comment]
Perfect! Best of both worlds.

**Comment by LeoNatan** [comment]
Out of scope for this project. Please feel free to submit PRs if you want to contribute.

**Comment by noomorph** [comment]
I would not necessarily agree that it is out of the scope, but I've checked the possible solutions and I can confidently say it can be done with compromises and only after configuration features land.

**Comment by LeoNatan** [comment]
The problem is we are spread too thin and too many enhancement requests. Feel free to open when you have time to implement.
